### DISCUSSION TITLE
Multiple guests

### DISCUSSION THREAD

**Discussion by mateferrox** [spatie/icalendar-generator#87] [Q&A]
Hello,
by chance there is the possibility of making a foreach, thus allowing to send invitations to a series of guests?
My code is as follows
![Immagine 2022-04-19 180526](https://user-images.githubusercontent.com/63313649/164047218-a157bf79-b459-47e3-ba90-b4a2de66b525.jpg)

**Comment by rubenvanassche** [comment]
Store your event in a variable like this:

```php
$event = Event::create('BLA')->....
```

And then:

```php
foreach($guests as $guest){
   $event->attendee(....)
}
```

**Reply by mateferrox** [reply]
Hi @rubenvanassche 
I tried to follow your suggestion (see image), but it gives me this error:
![1](https://user-images.githubusercontent.com/63313649/172924716-7bff7ed2-50f5-4712-9ee4-1dd380614fef.PNG)

Why?

**Reply by mateferrox** [reply]
@rubenvanassche any news?

**Reply by rubenvanassche** [reply]
Which error did you get? Seems valid PHP to me

**Reply by mateferrox** [reply]
@rubenvanassche the error is:
 Object of class Spatie\IcalendarGenerator\Components\Calendar could not be converted to string

**Reply by rubenvanassche** [reply]
Call `->get();` on it in the end

**Reply by mateferrox** [reply]
@rubenvanassche now new error:  Call to undefined method Spatie\IcalendarGenerator\Components\Event::get() 
![1](https://user-images.githubusercontent.com/63313649/179525331-d0069ccb-0c62-477e-8e0c-99c6a12a24e8.PNG)

**Reply by rubenvanassche** [reply]
You need to call this on your calendar not the event. Also do not forget to add the event to the calendar.

**Reply by mateferrox** [reply]
@rubenvanassche 
I added this piece to the code:
$evento->event($iCalendar)->get();
foreach ($attendee as $guest) {
$iCalendar->attendee($guest);
}

**Reply by rubenvanassche** [reply]
```php
foreach ($attendee as $guest) {
$iCalendar->attendee($guest);
}
$evento->event($iCalendar)->get();
```

**Reply by mateferrox** [reply]
@rubenvanassche I edit:
$iCalendar = Event::create($event->summary)
                    ->startsAt(new \DateTime($event->start->dateTime))
                    ->endsAt(new \DateTime($event->end->dateTime))
                    ->description($event->description)
                    ->address($event->location)
                    ->createdAt(new \DateTime('now'),'Europe/Rome')
                    ->organizer(strtolower(\Auth::user()->email),\Auth::user()->name)
                    ->alertMinutesBefore(60, 'L\'evento inizierà tra un\'ora');
            **foreach ($attendee as $guest) {
                $iCalendar->attendee($guest);
            }
            $evento->event($iCalendar)->get();**
But it continues to give me error:  Object of class Spatie\IcalendarGenerator\Components\Calendar could not be converted to string

**Reply by rubenvanassche** [reply]
What is the type of your `$guest` variable?
